tests/models/superpoint/test_modeling_superpoint.py

```diff
@@ -0,0 +1,287 @@
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in SUPERPOINT_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = SuperPointModel.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+
+def prepare_imgs():
+    image1 = Image.open("./tests/fixtures/tests_samples/COCO/000000039769.png")
+    image2 = Image.open("./tests/fixtures/tests_samples/COCO/000000004016.png")
+    return [image1, image2]
+
+
+@require_torch
+@require_vision
+class SuperPointModelIntegrationTest(unittest.TestCase):
+    @cached_property
+    def default_image_processor(self):
+        return AutoImageProcessor.from_pretrained("stevenbucaille/superpoint") if is_vision_available() else None
+
+    def infer_on_model(self, model):
+        preprocessor = self.default_image_processor
+        images = prepare_imgs()
+        inputs = preprocessor(images=images, return_tensors="pt").to(torch_device)
+
+        with torch.no_grad():
+            outputs = model(**inputs)
+
+        expected_number_keypoints_image0 = 568
+        expected_number_keypoints_image1 = 830
+        expected_max_number_keypoints = max(expected_number_keypoints_image0, expected_number_keypoints_image1)
+
+        expected_keypoints_shape = torch.Size((len(images), expected_max_number_keypoints, 2))
+        expected_scores_shape = torch.Size(
+            (
+                len(images),
+                expected_max_number_keypoints,
+            )
+        )
+        expected_descriptors_shape = torch.Size((len(images), expected_max_number_keypoints, 256))
+
+        # Check output shapes
+        self.assertEqual(outputs.keypoints.shape, expected_keypoints_shape)
+        self.assertEqual(outputs.scores.shape, expected_scores_shape)
+        self.assertEqual(outputs.descriptors.shape, expected_descriptors_shape)
+
+        expected_keypoints_image0_values = torch.tensor([[480.0, 9.0], [494.0, 9.0], [489.0, 16.0]]).to(torch_device)
+        expected_scores_image0_values = torch.tensor(
+            [0.0064, 0.0140, 0.0595, 0.0728, 0.5170, 0.0175, 0.1523, 0.2055, 0.0336]
+        ).to(torch_device)
+        expected_descriptors_image0_value = torch.tensor(-0.1096).to(torch_device)
+
+        # Check output values
+        self.assertTrue(torch.allclose(outputs.keypoints[0, :3], expected_keypoints_image0_values, atol=1e-4))
+        self.assertTrue(torch.allclose(outputs.scores[0, :9], expected_scores_image0_values, atol=1e-4))
+        self.assertTrue(torch.allclose(outputs.descriptors[0, 0, 0], expected_descriptors_image0_value, atol=1e-4))
+
+        # Check mask values
+        self.assertTrue(outputs.mask[0, expected_number_keypoints_image0 - 1].item() == 1)
+        self.assertTrue(outputs.mask[0, expected_number_keypoints_image0].item() == 0)
+        self.assertTrue(torch.all(outputs.mask[0, : expected_number_keypoints_image0 - 1]))
+        self.assertTrue(torch.all(outputs.mask[1]))
+
+    @slow
+    def test_inference(self):
```

Comment:
Let's just have a single `test_inference` test

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/25786/commits/573ad1913b10b27ffa733aeb90b7200d888a14fc)